File: src/main/scala/com/katus/chapter06/Dog.scala
package com.katus.chapter06

/**
 * @author SUN Katus
 * @version 1.0, 2020-08-19
 */
object Dog {   // 单例对象 类似于static 内存中只有一份
  val leg_num = 4

  def printSplitter(): Unit = {
    println("-" * 15)
  }
}

File: src/main/java/com/nx/vip/proxy/cglib/Dog.java
package com.nx.vip.proxy.cglib;

/**
 * @author SUN Katus
 * @version 1.0, 2021-08-09
 */
public class Dog implements IAnimal {
    @Override
    public final void run(String name) {
        System.out.println("Dog" + name + "----run");
    }

    @Override
    public void eat() {
        System.out.println("Dog----eat");
    }
}
